bsc-plugin/src/lib/rooibos/MockUtil.spec.ts

```diff
@@ -0,0 +1,387 @@
+                expect(program.getDiagnostics()).to.be.empty;
+                await builder.transpile();
+                let a = getContents('source/code.brs');
+                let b = trimLeading(`function __Person_builder()
+                instance = {}
+                instance.new = sub()
+                end sub
+                instance.sayHello = sub(a1, a2)
+                print "hello"
+                end sub
+                return instance
+                end function
+                function Person()
+                instance = __Person_builder()
+                instance.new()
+                return instance
+                end function`);
+                expect(a).to.equal(b);
+
+            });
+            it('will add stub code to namespace and global methods in a file with a class', async () => {
+                program.setFile('source/code.bs', `
+                namespace beings
+                class Person
+                    sub sayHello(a1, a2)
+                        print "hello"
+                    end sub
+                end class
+                function sayHello()
+                    print "hello2"
+                end function
+                end namespace
+                function sayHello()
+                    print "hello3"
+                end function
+            `);
+                program.validate();
+                expect(program.getDiagnostics()).to.be.empty;
+                await builder.transpile();
+                let a = getContents('source/code.brs');
+                let b = trimLeading(`function __beings_Person_builder()
+                instance = {}
+                instance.new = sub()
+                end sub
+                instance.sayHello = sub(a1, a2)
+                print "hello"
+                end sub
+                return instance
+                end function
+                function beings_Person()
+                instance = __beings_Person_builder()
+                instance.new()
+                return instance
+                end function
+
+                function beings_sayHello()
+                if RBS_SM_1_getMocksByFunctionName()["beings_sayhello"] <> invalid
+                result = RBS_SM_1_getMocksByFunctionName()["beings_sayhello"].callback()
+                return result
+                end if
+                print "hello2"
+                end function
+
+                function sayHello()
+                if RBS_SM_1_getMocksByFunctionName()["sayhello"] <> invalid
+                result = RBS_SM_1_getMocksByFunctionName()["sayhello"].callback()
+                return result
+                end if
+                print "hello3"
+                end function
+
+                function RBS_SM_1_getMocksByFunctionName()
+                if m._rMocksByFunctionName = invalid
+                m._rMocksByFunctionName = {}
+                end if
+                return m._rMocksByFunctionName
+                end function
+`);
```

Comment:
Why do you hate proper indentation? :P

Reply:
lol. I honestly find if I do anything else, it won't pass the test. Always been too busy to sort it out.